test/singa/test_cudnn_softmax.cc

```diff
@@ -33,27 +33,31 @@ TEST(CudnnSoftmax, Setup) {
 
   singa::LayerConf conf;
   singa::SoftmaxConf* softmaxconf = conf.mutable_softmax_conf();
-  softmaxconf->set_axis(2);
+  softmaxconf->set_algorithm("fast");
+  softmaxconf->set_mode("channel");
 
   sft.Setup(conf);
-  sft.InitCudnn(1, singa::kFloat32);
-  EXPECT_EQ(2, sft.Axis());
+  sft.InitCudnn(singa::Shape{1}, singa::kFloat32);
+  EXPECT_EQ(CUDNN_SOFTMAX_FAST, sft.Algorithm());
+  EXPECT_EQ(CUDNN_SOFTMAX_MODE_CHANNEL, sft.Mode());
 }
 
 TEST(CudnnSoftmax, Forward) {
-  const float x[] = {1.0f, 2.0f, 0.0f, -2.0f, -3.0f, -1.0};
+  const float x[] = {1.f, 2.f, 0.f, -2.f, -3.f, -1.f};
   size_t n = sizeof(x) / sizeof(float);
+  size_t batch = 2, c = 3;
   singa::CudaGPU cuda(0, 1);
-  singa::Tensor in(singa::Shape{n}, &cuda);
+  singa::Shape shape = {batch, c};
```

Comment:
I tested 1D first, and then changed to 2D.
Or maybe I can add separate tests for 1D and 2D.